sklearn/metrics/base.py

```diff
@@ -131,3 +132,94 @@ def _average_binary_score(binary_metric, y_true, y_score, average,
         return np.average(score, weights=average_weight)
     else:
         return score
+
+def _average_multiclass_score(binary_metric, y_true, y_score,
+                              average, multiclass):
+
+    """Uses the binary metric for multiclass classification
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples] or [n_samples, n_classes]
+        True multiclass labels
+
+    y_score : array, shape = [n_samples] or [n_samples, n_classes]
+        Target scores corresponding to probability estimates of a sample
+        belonging to a particular class
+
+    average : string, [None, 'macro' (default), 'weighted']
+        TODO: difference between 'macro' and None? Should there be both?
+        If ``None``, the scores for each class are returned. Otherwise,
+        this determines the type of averaging performed on the data:
+
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean.  This does not take label imbalance into account.
+        ``'weighted'``:
+            Calculate metrics for each label, taking into account the a priori
+            distribution of the classes.
+
+    binary_metric : callable, returns shape [n_classes]
+        The binary metric function to use.
+
+    Returns
+    -------
+    score : float or array of shape [n_classes]
+        If not ``None``, average the score, else return the score for each
+        classes.
+
+    """
+    average_options = (None, "macro", "weighted")
```

Comment:
This is a private function. If these arguments were already checked in the public function (which I think this was) we don't need to check it here again.
In particular, look at the code coverage of the test (after you implemented them ;).